data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqs/SqsEventProcessor.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.sqs;
+
+import org.opensearch.dataprepper.buffer.common.BufferAccumulator;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+
+import software.amazon.awssdk.services.sqs.model.Message;
+
+import java.io.IOException;
+    
+public class SqsEventProcessor {
+    private final SqsMessageHandler sqsMessageHandler;
+    SqsEventProcessor(final SqsMessageHandler sqsMessageHandler) {
+        this.sqsMessageHandler= sqsMessageHandler;
+    }
+
+    void addSqsObject(final Message message,
+                      final String url,
+                      final BufferAccumulator<Record<Event>> bufferAccumulator,
```

Comment:
BufferAccumulator is not thread safe. I see that you are creating BufferAccumulator in each SqsWorker thread which is good. Just that I am not sure event processor (or the message handler) becomes another thread? if not, then we are good. If yes then you need make sure that it is not creating any synchronization issues here.

Reply:
As you've said, I created each BufferAccumulator in each SqsWorker thread. Within each SqsWorker, no new threads are created, so concurrency shouldn't be an issue here. However, please let me know if my understanding is incorrect here.